app/datarequest/add.js

```diff
@@ -111,6 +112,17 @@ const fields = {
   DataSelection: DataSelectionTable
 };
 
+function validate(formData, errors) {
```

Comment:
Can we add this restriction to the schema using oneOf or anyOf?

Reply:
I tried. Can't get it to work. And even if I could, part of the desired behaviour (showing all options at once) doesn't seem to be supported: https://github.com/rjsf-team/react-jsonschema-form/issues/1348.

Unless someone else can get it to work, I think the current solution will have to do.

Reply:
Another possible route (thanks @reavdhoef) would be to add:

```
      "not": {
        "properties": {
          "contribution_time": { "enum": ["No"] },
          "contribution_financial": { "enum": ["No"] },
          "contribution_favor": { "enum": ["No"] }
        }
      },
```

That works, but the error message is useless: "should NOT be valid". Changing the error message is not trivial (https://github.com/ajv-validator/ajv/issues/573), the suggested solution in that issue is to install the `ajv-errors` package and use that to customize the message, though the issue has been resolved for `oneOf`, so perhaps also for `not`. @reavdhoef is going to take a shot at fixing the error message.

Reply:
I am taking too long to figure out how to deploy a new version of add.js (I have no clue how to build a new version for in the public folder), but the way to go is to transform the error message on top of the 'not'-solution illustrated above. This can be done easily with the transformErrors property on the Form, see [the docs](https://react-jsonschema-form.readthedocs.io/en/latest/usage/validation/#custom-error-messages) .

In the example given, every error is checked for its name. In this case I suggest to use the following transform function:

```
function transformErrors(errors) {
    return errors.map(error => {
        if(error.name === "not" && error.property === ".contribution") {
            error.message = "At least one contribution has to be made"
        }
        return error;
    });
}
```

Reply:
That works. Nice.